app/assets/javascripts/controllers/deploy_groups_ctrl.js

```diff
@@ -1,15 +1,46 @@
 samson.controller('DeployGroupsCtrl', function($scope, $http, $location, $window) {
+  var MAX_PAGES = 20;
 
   function init() {
-    $http.get($location.path() + '.json').success(function(result) {
-      result.deploys.forEach(function(deploy) {
-        deploy.start = new Date(deploy.started_at || deploy.created_at);
+    getMoreDeploys(1);
+  }
+
+  function getMoreDeploys(page) {
+    if (page < MAX_PAGES) {
+      $http.get($location.path() + '.json?page=' + page).success(function(result) {
+        if (result.deploys.length > 0) {
+          result.deploys.forEach(function(deploy) {
+            deploy.start = new Date(deploy.started_at || deploy.created_at);
+          });
+
+          $scope.items.add(result.deploys);
+          $scope.deployIds = $scope.deployIds.concat(_.pluck(result.deploys, 'id'));
+
+          adjustVisibleItems(result.deploys);
+          getMoreDeploys(page + 1);
```

Comment:
Yea, this will stop when no more deploys are returned by the server or the MAX_PAGES is reached.